presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,311 @@
+    {
+        PartitionFunction partitionFunction = new LocalPartitionGenerator(new InterpretedHashGenerator(ImmutableList.of(BIGINT), new int[] {0}), PARTITION_COUNT);
+        PagesSerdeFactory serdeFactory = new PagesSerdeFactory(new BlockEncodingManager(new TypeRegistry()), false);
+        PartitionedOutputBuffer buffer = createPartitionedOutputBuffer();
+        PartitionedOutputOperator.PartitionedOutputFactory operatorFactory;
+        if (shouldReplicate) {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty()),
+                    true,
+                    OptionalInt.of(0),
+                    buffer,
+                    PARTITION_MAX_MEMORY);
+            return (PartitionedOutputOperator) operatorFactory
+                    .createOutputOperator(0, new PlanNodeId("plan-node-0"), REPLICATION_TYPES, Function.identity(), serdeFactory)
+                    .createOperator(createDriverContext());
+        }
+        else {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty(), Optional.empty()),
+                    false,
+                    OptionalInt.empty(),
+                    buffer,
+                    PARTITION_MAX_MEMORY);
+            return (PartitionedOutputOperator) operatorFactory
+                    .createOutputOperator(0, new PlanNodeId("plan-node-0"), TYPES, Function.identity(), serdeFactory)
+                    .createOperator(createDriverContext());
+        }
+    }
+
+    private PartitionedOutputBuffer createPartitionedOutputBuffer()
+    {
+        OutputBuffers buffers = createInitialEmptyOutputBuffers(PARTITIONED);
+        for (int partition = 0; partition < PARTITION_COUNT; partition++) {
+            buffers = buffers.withBuffer(new OutputBuffers.OutputBufferId(partition), partition);
+        }
+        PartitionedOutputBuffer buffer = createPartitionedBuffer(
+                buffers.withNoMoreBufferIds(),
+                new DataSize(Long.MAX_VALUE, BYTE)); // don't let output buffer block
+        buffer.registerLifespanCompletionCallback(ignore -> {});
+
+        return buffer;
+    }
+
+    private Page createSimplePage()
+    {
+        return new Page(buildBigIntBlock(POSITION_COUNT, false));
+    }
+
+    private Page createPageWithDictionary()
+    {
+        return new Page(buildDictionaryBlock(buildBigIntBlock(POSITION_COUNT, false), POSITION_COUNT));
+    }
+
+    private Page createPageWithRunLength()
+    {
+        return new Page(buildRleBlock(buildBigIntBlock(POSITION_COUNT, false), POSITION_COUNT));
+    }
+
+    private Page createPageWithNullBlock()
```

Comment:
- add a constant for null block
- use `Page(int positionCount, Block... blocks)` constructor
- inline this method and the next 2 methods

```
private static final Block NULL_BLOCK = new RunLengthEncodedBlock(BIGINT.createBlockBuilder(null, 1).appendNull().build(), POSITION_COUNT);;
...
        for (int i = 0; i < NUM_OF_ITERATIONS; i++) {
            partitionedOutputOperator.addInput(new Page(POSITION_COUNT, NULL_BLOCK, createLongSequenceBlock(0, POSITION_COUNT)));
        }

        for (int i = 0; i < NUM_OF_ITERATIONS; i++) {
            partitionedOutputOperator.addInput(new Page(POSITION_COUNT, NULL_BLOCK, createLongDictionaryBlock(0, POSITION_COUNT)));
        }

        for (int i = 0; i < NUM_OF_ITERATIONS; i++) {
            partitionedOutputOperator.addInput(new Page(POSITION_COUNT, NULL_BLOCK, createRLEBlock(RANDOM.nextLong(), POSITION_COUNT)));
        }
```